arcgis-ios-sdk-samples/Scenes/Calculate 3D distance/Calculate3DDistanceViewController.swift

```diff
@@ -0,0 +1,283 @@
+        let cartesian2 = cartesian(fromSpherical: spherical2)
+        
+        let distance = Measurement(
+            value: simd_distance(cartesian1, cartesian2),
+            unit: UnitLength.meters
+        )
+        distanceLabel.text = String(format: "Distance: %@", measurementFormatter.string(from: distance))
+    }
+    
+    /// Converts a spherical point to a cartesian point.
+    ///
+    /// - Parameter spherical: A point in the spherical coordinate system.
+    /// - Returns: A point in the cartesian coordinate system.
+    func cartesian(fromSpherical spherical: simd_double3) -> simd_double3 {
+        let earthRadius = Measurement(value: 6_371_000, unit: UnitLength.meters)
+        
+        let latitude = Measurement<UnitAngle>(value: spherical.x, unit: .degrees)
+            .converted(to: .radians)
+        let longitude = Measurement<UnitAngle>(value: spherical.y, unit: .degrees)
+            .converted(to: .radians)
+        let radius = earthRadius + Measurement(value: spherical.z, unit: .meters)
+        return simd_double3(
+            earthRadius.value * cos(latitude.value) * cos(longitude.value),
+            earthRadius.value * cos(latitude.value) * sin(longitude.value),
+            radius.value * sin(latitude.value)
+        )
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["Calculate3DDistanceViewController"]
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        drawStatusObservation = sceneView.observe(\.drawStatus, options: .initial) { [weak self] (sceneView, _) in
+            DispatchQueue.main.async {
+                guard sceneView.drawStatus == .completed,
+                    let self = self else {
+                        return
+                }
+                // Start the animation.
+                self.startAnimatingGraphics()
+                self.drawStatusObservation = nil
+            }
+        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        stopAnimatingGraphics()
+    }
+}
+
+/// A point in a two-dimensional coordinate system.
+struct Point: Equatable {
+    /// The x-coordinate of the point.
+    var x: Double
+    /// The y-coordinate of the point.
+    var y: Double
+}
+
+/// A function of a single variable.
+protocol Function {
+    /// Applies the function to a given value.
+    ///
+    /// - Parameter x: A value.
+    /// - Returns: The result of applying the function to `x`.
+    func apply(to x: Double) -> Double
+}
+
+/// An absolute value function of the form `y=a|x-h|+k`. The vertex of the
+/// function is at `(h, k)`.
+struct AbsoluteValueFunction: Function {
```

Comment:
I don't really have a suggestion here but I wouldn't mind if it was named more for what it does in terms of calculating position, rather than how it does it.  That being said, this name gets hidden when the assignments to `redPositionFunction` and `greenPositionFunction` are made so I'd be okay if it didn't change.